### PR TITLE
feat(chdb): implement chdb - the embedded clickhouse - backend

### PR DESCRIPTION
**Pull request by kszucs**
A big chunk of the backends testing suite is already passing, but there is still work to do:
1. Support passing in memory tables, this can be done using the same "pass as a file" trick used in [chdb.dataframe](https://github.com/chdb-io/chdb/blob/main/chdb/dataframe/query.py)
2. Better error handling
3. Pass the right dialect everywhere

Test suite is currently at `247 failed, 1067 passed, 28 skipped, 29854 deselected, 131 xfailed, 6 errors`

### CODE REVIEW

File: ibis/backends/chdb/__init__.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            Backend specific query arguments
+
+        Returns
+        -------
+        Cursor
+            Clickhouse cursor
+
+        """
+        with contextlib.suppress(AttributeError):
+            query = query.sql(dialect="clickhouse", pretty=True)
```
Comment (cpcloud) [inline ibis/backends/chdb/__init__.py:62]:
```suggestion
            query = query.sql(dialect=self.dialect, pretty=True)
```

File: ibis/backends/chdb/__init__.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        """
+        if schema is not None:
+            raise com.UnsupportedBackendFeatureError(
+                "`schema` namespaces are not supported by chdb"
+            )
+        query = sge.Describe(this=sg.table(table_name, db=database))
+        with self._safe_raw_sql(query, fmt="arrowtable") as table:
+            names = table.column("name").to_pylist()
+            types = table.column("type").to_pylist()
```
Comment (cpcloud) [inline ibis/backends/chdb/__init__.py:98]:
```suggestion
            names = table["name"].to_pylist()
            types = table["type"].to_pylist()
```

File: ibis/backends/chdb/__init__.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        """Execute an expression."""
+        table = expr.as_table()
+        sql = self.compile(table, **kwargs)
+        self._log(sql)
+
+        external_tables = self._collect_in_memory_tables(expr, external_tables)
+        if external_tables:
+            raise NotImplementedError("External tables are not yet supported")
+
+        df = self.con.query(sql, "arrowtable").to_pandas()
```
Comment (cpcloud) [inline ibis/backends/chdb/__init__.py:117]:
```suggestion
        df = self.con.query(sql, fmt="arrowtable").to_pandas()
```

File: ibis/backends/chdb/__init__.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    ):
+        table = expr.as_table()
+        sql = self.compile(table, **kwargs)
+        self._log(sql)
+
+        external_tables = self._collect_in_memory_tables(expr, external_tables)
+        if external_tables:
+            raise NotImplementedError("External tables are not yet supported")
+
+        table = self.con.query(sql, "arrowtable")
```
Comment (cpcloud) [inline ibis/backends/chdb/__init__.py:135]:
```suggestion
        table = self.con.query(sql, fmt="arrowtable")
```

File: ibis/backends/chdb/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    supports_json = False
+    force_sort = True
+    rounding_method = "half_to_even"
+    deps = ("chdb",)
+
+    @property
+    def native_bool(self) -> bool:
+        return True
+        # [(value,)] = self.connection.con.query("SELECT true").result_set
+        # return isinstance(value, bool)
```
Comment (cpcloud) [inline ibis/backends/chdb/tests/conftest.py:35]:
```suggestion
```

We definitely will not be supporting versions of chdb that don't support native booleans.

File: ibis/backends/chdb/tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+    @property
+    def test_files(self) -> Iterable[Path]:
+        return self.data_dir.joinpath("parquet").glob("*.parquet")
+
+    @property
+    def ddl_script(self) -> Iterable[str]:
+        parquet_dir = self.data_dir / "parquet"
+        for sql in super().ddl_script:
+            yield sql.format(parquet_dir=parquet_dir)
```
Comment (cpcloud) [inline ibis/backends/chdb/tests/conftest.py:45]: I'd rather not introduce templating if we can avoid it. How about hard coding the path to `ci/ibis-testing-data/...` like we do with everything else?

File: ibis/backends/chdb/tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        Parameters
+        ----------
+        data_dir
+            Location of test data
+        script_dir
+            Location of scripts defining schemas
+        """
+        con = self.connection.con
+
+        con.query(f"CREATE DATABASE {database} ENGINE = Atomic")
```
Comment (cpcloud) [inline ibis/backends/chdb/tests/conftest.py:59]: Does this work with our process-based parallel testing setup (i.e., `pytest-xdist`)?

File: ibis/config.py [hunk 8]
```diff
@@ -175,6 +175,7 @@ class Options(Config):
     default_backend: Optional[Any] = None
     context_adjustment: ContextAdjustment = ContextAdjustment()
     sql: SQL = SQL()
+    chdb: Optional[Config] = None
```
Comment (cpcloud) [inline ibis/config.py:178]: Do we really need this?

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: ibis/backends/chdb/tests/conftest.py (hunk 7 above)
Comment (kszucs) [reply ibis/backends/chdb/tests/conftest.py:57]: Not there yet :)

**Comment by kszucs** [comment]
I won't have time to keep working on this in the near future, so closing it.

**Comment by zhenzhongxu** [comment]
@auxten we've reopened the PR based on your feedback. 
@kszucs, we'd like to move the Ibis/chDB integration forward. I'm wondering if you have any useful context to whoever will pick this up.

**Comment by jcrist** [comment]
Going to reclose this for now. Can always reopen if we want to continue the work, but no need to keep it open if we're not hacking on it proactively.

**Pull request closed without merging**